ci/steps/getlogs.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+#  export KUBECONFIG=/Users/d058463/Documents/SAP-Dev/Kubernetes/kubeconfigs/kubeconfig--gdnlinux--build.yaml
+from collections import namedtuple
+from dataclasses import dataclass
+import os
+from typing import Dict
+import zipfile
+
+from kubernetes import client, config
+
+
+def getlogs(repo_dir: str, namespace: str, pipeline_run_name: str):
+    @dataclass
+    class TaskRunInfo:
+        name: str
+        pod_name: str
+        steps: Dict[str, str]  # step-name --> step container
+
```

Comment:
consider moving this to the model

Reply:
also, an "anonymous" class is a bit unusual. Consider moving it out of the function

Reply:
I prefer this one to keep as it is. This dataclass is convenient but exactly used only in this function. Why make it global?